packages/fuel-gauge/test/fixtures/forc-projects/options/src/main.sw

```diff
@@ -0,0 +1,46 @@
+contract;
+
+enum OptionEnum {
+    a: Option<u8>,
+    b: Option<u16>,
+}
+
+struct OptionStruct {
+    one: OptionEnum,
+    two: Option<u32>,
+}
+
+abi OptionContract {
+    fn echo_option(arg: Option<u8>) -> Option<u8>;
+    fn echo_struct_enum_option(arg: OptionStruct) -> OptionStruct;
+    fn echo_vec_option(arg: Vec<Option<u32>>) -> Vec<Option<u32>>;
+    fn echo_tuple_option(arg: (Option<u8>, Option<u16>)) -> (Option<u8>, Option<u16>);
+    fn echo_enum_option(arg: OptionEnum) -> OptionEnum;
+    fn echo_array_option(arg: [Option<u16>; 3]) -> [Option<u16>; 3];
+}
+
+impl OptionContract for Contract {
+    fn echo_option(arg: Option<u8>) -> Option<u8> {
```

Comment:
I think we should have tested for the internal array of `None` case. 

```rust
fn echo_array_option() -> [OptionEnum; 3] {
       [None; 3]
}

Reply:
It's a good point but `undefined` when encoded as `Option<u8>` == `None`.

Take a look at the [OptionCoder.toSwayOption()](https://github.com/FuelLabs/fuels-ts/blob/63a9d55a18e8b16bd10edda5535610c118770765/packages/abi-coder/src/encoding/coders/OptionCoder.ts#L14-L19) where `undefined` is converted to `None` before being encoded by an `EnumCoder`.

I usually start with a print test to just test byte decoding and to see how we expect the bytes to be arranged:

```rs
fn print_option() -> Option<u8> {
    Option::None
}
```

and then will move to an echo test for encode and decode:

```rs
fn echo_option(arg: Option<u8>) -> Option<u8> {
    arg
}
```

Which is why most of our type tests are echoes rather than prints. So then we can use TS to represent Sway values correctly. So IMO this [test](https://github.com/FuelLabs/fuels-ts/pull/2322/files#diff-5118910301c29105538f3849ae3483c491fd11733ea5d7d78f4ad178a5ff905aR135-R141) does do the job.

Reply:
Good point, noted